image_segmentation/pytorch/capture_profile.py

```diff
@@ -0,0 +1,43 @@
+import argparse
+
+import torch_xla.debug.profiler as xp
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Performs an on-demand profiling session on provided profiler servers."
+    )
+
+    parser.add_argument(
+        "--service_addr",
+        dest="service_addr",
+        type=str,
+        required=True,
+        help='comma delimited string of addresses of the profiling servers to profile. ex. "10.0.0.2:8466" or "localhost:9012".',
+    )
+    parser.add_argument(
+        "--logdir",
+        dest="logdir",
+        type=str,
+        required=True,
+        help='the path to write profiling output to. Both the profiler client and server must have access. ex. "gs://bucket/file/path".',
+    )
+    parser.add_argument(
+        "--duration_ms",
+        dest="duration_ms",
+        type=int,
+        default=30000,
+        help="duration in milliseconds for tracing the server.",
+    )
+
+    flags = parser.parse_args()
+
+    xp.trace(
```

Comment:
Suggesting two additional flags:

+ an optional flag to wait for the user to trigger a profile capture. Something like `start_profile = input("Press any key to start profiling\n")`

+ an optional flag to offer the user an opportunity to capture multiple such profiles. Essentially, put the `input` command in a `for` loop.

Reply:
Thanks for the great suggestions.

I have added a single flag `--interactive` that does what you said:

* When `--interactive=once`, the profiler client asks for user confirmation before starting profiling.
* When `--interactive=loop`, the profiler client repeatedly runs profiling, asking for user confirmation on each run.
* `--interactive` defaults to `None`, which disables interactive mode.

I have tested `capture_profile.py` with the `--interactive` flag against UNet3D and confirmed it works as expected.